File: scratch/inject_lang.py
import os
import re

def update_forms():
    root_dir = r"c:\Users\User\Desktop\Web Design Google\Turpone Foods"
    
    # Regex to find the <form> start tag
    form_regex = re.compile(r'(<form[^>]*action="https://script\.google\.com[^>]*>)')
    
    for dirpath, dirnames, filenames in os.walk(root_dir):
        if 'node_modules' in dirpath or '.git' in dirpath:
            continue
            
        for filename in filenames:
            if filename.endswith('.html'):
                filepath = os.path.join(dirpath, filename)
                rel_path = os.path.relpath(filepath, root_dir).replace('\\', '/')
                
                # Determine language
                lang = "en"
                if rel_path.startswith("fr/"):
                    lang = "fr"
                elif rel_path.startswith("es/"):
                    lang = "es"
                
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Check if it has a form
                if form_regex.search(content):
                    # Check if already has a lang hidden input
                    if 'name="lang"' not in content:
                        # Inject the hidden input right after the form opening tag
                        new_content = form_regex.sub(rf'\1<input name="lang" type="hidden" value="{lang}"/>', content)
                        
                        with open(filepath, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Updated: {rel_path} with lang={lang}")

if __name__ == "__main__":
    update_forms()

File: scratch/inject_lang.js
const fs = require('fs');
const path = require('path');

function getHtmlFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const filePath = path.join(dir, file);
    if (fs.statSync(filePath).isDirectory()) {
      if (file !== 'node_modules' && file !== '.git') {
        getHtmlFiles(filePath, fileList);
      }
    } else if (file.endsWith('.html')) {
      fileList.push(filePath);
    }
  }
  return fileList;
}

const rootDir = "c:/Users/User/Desktop/Web Design Google/Turpone Foods";
const htmlFiles = getHtmlFiles(rootDir);
const formRegex = /(<form[^>]*action="https:\/\/script\.google\.com[^>]*>)/;

for (const file of htmlFiles) {
  const relPath = path.relative(rootDir, file).replace(/\\/g, '/');
  
  let lang = "en";
  if (relPath.startsWith("fr/")) lang = "fr";
  else if (relPath.startsWith("es/")) lang = "es";
  
  let content = fs.readFileSync(file, 'utf8');
  if (formRegex.test(content) && !content.includes('name="lang"')) {
    content = content.replace(formRegex, `$1<input name="lang" type="hidden" value="${lang}"/>`);
    fs.writeFileSync(file, content, 'utf8');
    console.log(`Updated: ${relPath} with lang=${lang}`);
  }
}
